Mobile/ShopLens/Droid/Activities/ShoppingCartActivity.cs

```diff
@@ -72,31 +95,47 @@ private bool IsTalkBackEnabled()
 
         private void InitiateNoTalkBackMode()
         {
-            tts = new TextToSpeech(this, this);
-
+            needUserAnswerId = ConfigurationManager.AppSettings["AnswerUtteranceId"];
+            askUserToRepeat = ConfigurationManager.AppSettings["AskToRepeat"];
+            afterActionAsk = ConfigurationManager.AppSettings["AfterActionMsg"];
+            cmdOpenMain = ConfigurationManager.AppSettings["CmdOpenMain"];
+            cmdOpenList = ConfigurationManager.AppSettings["CmdOpenList"];
+            cmdVoiceCart = ConfigurationManager.AppSettings["CmdVoiceCartList"];
+            cmdHelp = ConfigurationManager.AppSettings["CmdHelp"];
+            cmdRemind = ConfigurationManager.AppSettings["CmdRemind"];
+
+            shopLensTTS = new ShopLensTextToSpeech(this, TTSSpeakAfterInit, TTStoppedSpeaking);
             voiceRecognizer = new ShopLensSpeechRecognizer(OnVoiceRecognitionResults);
         }
 
-        public void OnInit([GeneratedEnum] OperationResult status)
+        private void TTSSpeakAfterInit(object sender, EventArgs e)
         {
-            if (status == OperationResult.Error)
-            {
-                tts.SetLanguage(Locale.Default);
-            }
-            else if (status == OperationResult.Success)
+            var message = ConfigurationManager.AppSettings["CartOnRestartMsg"];
+            shopLensTTS.Speak(message, needUserAnswerId);
+        }
+
+        private void TTStoppedSpeaking(object sender, UtteranceIdArgs e)
+        {
+            if (e.UtteranceId == needUserAnswerId)
             {
-                tts.SetLanguage(Locale.Us);
+                voiceRecognizer.RecognizePhrase(this);
             }
         }
 
+        private void AskAfterAction()
+        {
+            shopLensTTS.Speak(afterActionAsk, needUserAnswerId);
+        }
+
         private void SpeakOut(string message, int checkDelay)
         {
             if (!string.IsNullOrEmpty(message))
             {
-                tts.Speak(message, QueueMode.Flush, null, null);
+                shopLensTTS.Speak(message, null);
 
-                while (tts.IsSpeaking)
+                while (shopLensTTS.tts.IsSpeaking)
                 {
+                    // TODO: Tomas needs to fix this.
```

Comment:
I left this method so that Tomas could figure out, how to get rid of that while statement.